src/core/config/config.go

```diff
@@ -480,3 +480,9 @@ func QuotaSetting() (*models.QuotaSetting, error) {
 		StoragePerProject: cfgMgr.Get(common.StoragePerProject).GetInt64(),
 	}, nil
 }
+
+// GetPermittedRegistryTypesForProxyCache returns the permitted registry types for proxy cache
+func GetPermittedRegistryTypesForProxyCache() []string {
+	types := os.Getenv("PERMITTED_REGISTRY_TYPES_FOR_PROXY_CACHE")
```

Comment:
Setting in configure is better than env variable, because in tile env, if user need to modify the setting, change in the env cannot persistent, only configure in database can persistent in DB and can be migrated during upgrade.

Reply:
If we want to allow users to customize it, it will be exposed in harbor.yml, such that the env can be persisted

Reply:
The system configuration isn't persisted in the database